contracts/src/v0.8/automation/dev/v2_2/AutomationRegistryBase2_2.sol

```diff
@@ -953,13 +951,36 @@ abstract contract AutomationRegistryBase2_2 is ConfirmedOwner, ExecutionPreventi
     }
   }
 
+  /**
+   * @notice Returns true if the registry is paused, false otherwise
+   */
+  function _isRegistryPaused() internal view returns (bool) {
+    return (s_hotVars.boolFlags & (1 << 0)) != 0;
+  }
+
+  /**
+   * @notice Returns true if the registry is reentrancy guarded, false otherwise
+   */
+  function _isReentrancyGuarded() internal view returns (bool) {
+    return (s_hotVars.boolFlags & (1 << 1)) != 0;
+  }
+
+  /**
+   * @notice Returns true if the registry enabled reorg protection, false otherwise
+   */
+  function _isReorgProtectionEnabled() internal view returns (bool) {
+    return (s_hotVars.boolFlags & (1 << 2)) != 0;
+  }
```

Comment:
all of these functions do a state read (this is why we use the `s_` prefix for state variables, to make state reads obvious). Meaning if we call these three functions in the same tx, they will _all_ pay 2000 gas to read the _same_ variable. Not a good use of gas! This will actually _increase_ the total gas needed for a tx, and we wanted to combine these booleans into one in order to save gas!

A simple way to fix this is to accept the `boolFlags` param as an argument. This way we can read it once from state in `transmit()` and then pass it to each function when needed.